### DISCUSSION TITLE
Guidance on GSOC 2025

### DISCUSSION THREAD

**Discussion by Adarsh2345** [alan-turing-institute/sktime#7355] [Q&A] [answered]
Hey there!
I'm Adarsh and I found sktime through brushing over last year's GSOC organizations list and I've found your past projects to be very interesting.
I currently have completed few projects on supervised and unsupervised machine learning using Scikit-Learn and TensorFlow.
Any specific skills/technologies that I need to master so that I can start making contributions to sktime?

**Comment by fkiraly** [comment, marked as answer]
@Adarsh2345, you are very early! Organisations themselves apply only in January, so we do not know yet whether `sktime` will be part of GSoC 2025 - although I would say there is a good chance given our track record.

The contributing guide is here: https://www.sktime.net/en/latest/get_involved/contributing.html#contributing

your background seems reasonable on high-level, I would recommend to pick a good first issue or two.

You can find our GSoC registration and selection process here: https://github.com/sktime/mentoring/blob/main/internships/gsoc2024.md, this requires having already made at least a small pull request, which is used in evaluation.

Most likely, the process will stay the same for 2025, and only the dates and timelines will change.

**Reply by Adarsh2345** [reply]
Thank you so much for your response, @fkiraly!!

**Comment by RATED-R-SUNDRAM** [comment]
Hi @fkiraly  , I'm shivam sundram , I've working experience and domain knowledge about data science and python, I'm new to open source contributing and I saw this org participated in GSOC last year, I'm looking forward to GSOC this year and looking for a head-start  where i can start contributing in this org, Any suggestion guidance if you can provide

**Reply by fkiraly** [reply]
Sure!

The contributing guide is here: https://www.sktime.net/en/latest/get_involved/contributing.html#contributing

Here is an issue with suggested starter items: https://github.com/sktime/sktime/issues/1147

**Comment by rishavtarway** [comment]
Hi @fkiraly , I am Rishav Tarway 2nd Year Btech Undergrad pursuing  CS with keen interest in python sphinx and scikit-learn looking over last year project in gsoc from sktime which were too amazing to work on wanted to ask if you have projects for this year any idea list or participating in this year 

Also one thing to highlight the Communication Channel for sktime says **invalid** so please update so who are interested can join can have their discussion over this amazing repository

**Reply by fkiraly** [reply]
> if you have projects for this year any idea list or participating in this year

yes, this is also mandatory for GSoC. See links below.

> Also one thing to highlight the Communication Channel for sktime says invalid so please update so who are interested can join can have their discussion over this amazing repository

Can you provide an exact link that is broken, so we can fix it?

**Reply by rishavtarway** [reply]
Now i have tried it on different browser its working over there the communication channel discord link for sktime .

**Comment by fkiraly** [comment]
Small update:

* the project list for 2025 is being compiled here: https://github.com/sktime/mentoring/blob/main/internships/gsoc2025.md, this file will also contain the final version.
* a prioritized high-level roadmap for 2025 is available here: https://github.com/sktime/sktime/issues/7707
